project/src/test/java/com/google/sps/SearchDataServletTest.java

```diff
@@ -58,45 +131,98 @@ public void setUp() throws Exception {
     getRequest = Mockito.mock(HttpServletRequest.class);
     getResponse = Mockito.mock(HttpServletResponse.class);
 
-    stringWriter = new StringWriter();
-    printWriter = new PrintWriter(stringWriter);
-    Mockito.when(getResponse.getWriter()).thenReturn(printWriter);
+    getStringWriter = new StringWriter();
+    getPrintWriter = new PrintWriter(getStringWriter);
+    Mockito.when(getResponse.getWriter()).thenReturn(getPrintWriter);
+
+    postStringWriter = new StringWriter();
+    postPrintWriter = new PrintWriter(postStringWriter);
+    Mockito.when(postResponse.getWriter()).thenReturn(postPrintWriter);
+
+    secondPostStringWriter = new StringWriter();
+    secondPostPrintWriter = new PrintWriter(secondPostStringWriter);
+    Mockito.when(secondPostResponse.getWriter()).thenReturn(secondPostPrintWriter);
 
     searchDataServlet = new SearchDataServlet();
+    eventCreationServlet = new EventCreationServlet();
+    mockKeywordHelper = Mockito.mock(KeywordHelper.class);
+    eventCreationServlet.setSearchStore(new SearchStore(mockKeywordHelper));
+
+    // Mock a request to trigger the SpannerClient setup to run
+    MockServletContext mockServletContext = new MockServletContext();
+    new SpannerClient().contextInitialized(new ServletContextEvent(mockServletContext));
+    SpannerTestTasks.setup();
+    authenticationHelper.setUp();
+
+    SpannerTasks.insertOrUpdateUser(HOST);
+    loginHost();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    SpannerTestTasks.cleanup();
+    authenticationHelper.tearDown();
   }
 
   @Test
-  public void addEventAndRetrieveResultsForKeywordNotInEventDescription_noResultsReturned()
+  public void oneEvent_KeywordNotRelevantInEventTitleOrDescription_noResultsReturned()
```

Comment:
@hlz03 @liangqigong a best practices question: generally would it be better for tests to name our tests descriptively like this vs writing a comment to describe it?

Reply:
I think this name should be fine. And if you think it is to hard describe it in a few words, a comment is also acceptable.